CoreScriptsRoot/Modules/Server/ServerChat/ChatServiceRunner.lua

```diff
@@ -294,6 +297,21 @@ local systemChannel = ChatService:AddChannel("System")
 allChannel.Leavable = false
 allChannel.AutoJoin = true
 
+allChannel:RegisterGetWelcomeMessageFunction(function(speaker)
+	if RunService:IsStudio() then
+		return
+	end
+	local player = speaker:GetPlayer()
+	if player then
+		local success, canChat = pcall(function()
+			return Chat:CanUserChatAsync(player.UserId)
+		end)
+		if success and not canChat then
+			return "Enable in-game chat in your privacy settings to be able to chat."
```

Comment:
This seems like a call for players to change their privacy settings? Do we want to do this?